### PR TITLE
Update AI endpoint to enable measuring costs

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #10214

## Description

- Replaces `jetpack-ai/completions` with `text-completion` endpoint
- Adds a way to fetch JWT token.
- Updates unit tests.

pe5sF9-1Il-p2

Although the `text-completion` endpoint can be accessed by self-hosted stores with Jetpack/JCP, we are not making AI features available to stores hosted outside WPCOM. Decision - pe5sF9-1Il-p2#comment-2501

## Testing instructions

**WPCOM store**
- Login into using a store hosted on WPCOM (Free trial/Paid)
- Navigate to Products tab and open a product
- Tap "Write with AI" and try generating content
- Ensure that AI generation works as before
- Dismiss the description sheet and tap "Share" button
- From the presented share sheet tap "Write with AI" and try generating share message
- Ensure that AI generation works as before

**Other stores**
- Login into a non-WPCOM store
- Navigate to Products tab and open a product
- Ensure that you don't see "Write with AI" button
- Dismiss the description sheet and tap "Share" button
- Ensure that you don't see "Write with AI" button in the share sheet

## Screenshots

**Product form**
| WPCOM store | Other Stores |
|--------|--------|
| Simulator Screen Shot - iPhone 14 Pro - 2023-07-12 at 17 27 27 | Simulator Screen Shot - iPhone 14 Pro - 2023-07-12 at 17 36 44 |

**Product sharing message**
| WPCOM store | Other Stores |
|--------|--------|
| Simulator Screen Shot - iPhone 14 Pro - 2023-07-12 at 17 28 43 | Simulator Screen Shot - iPhone 14 Pro - 2023-07-12 at 17 37 41 |

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jaclync** [comment]
@selanthiraiyan 👋 I thought it's better to wait for product confirmation in pe5sF9-1Il-p2 before reviewing & merging the PR since it can have a significant impact on the AI usage and potentially the cost.

**Comment by selanthiraiyan** [comment]
I agree, Jaclyn! I am marking this as a draft until we get confirmation.

File: Networking/Networking/Remote/GenerativeContentRemote.swift
Comment (itsmeichigo) [inline Networking/Networking/Remote/GenerativeContentRemote.swift]: Nit: Can we rename this to something more friendly like `tokenNotFound`?

File: Networking/NetworkingTests/Remote/GenerativeContentRemoteTests.swift
Comment (itsmeichigo) [inline Networking/NetworkingTests/Remote/GenerativeContentRemoteTests.swift]: ❓ Is this redundant? If not, could you add a comment explaining why we must repeat this step?

File: Networking/NetworkingTests/Remote/GenerativeContentRemoteTests.swift
Comment (itsmeichigo) [inline Networking/NetworkingTests/Remote/GenerativeContentRemoteTests.swift]: Same question as above.

**Review by itsmeichigo — APPROVED** [review]
LGTM 👍 :shipit:

File: Networking/Networking/Remote/GenerativeContentRemote.swift
Comment (selanthiraiyan) [reply Networking/Networking/Remote/GenerativeContentRemote.swift]: Done in https://github.com/woocommerce/woocommerce-ios/pull/10218/commits/34952eb23a10b461ae1c61580345dac3e3b3008a

File: Networking/NetworkingTests/Remote/GenerativeContentRemoteTests.swift
Comment (selanthiraiyan) [reply Networking/NetworkingTests/Remote/GenerativeContentRemoteTests.swift]: This checks that JWT is not requested again until "invalid token error" is received. Added comment in https://github.com/woocommerce/woocommerce-ios/pull/10218/commits/1a1561278eae0f2e4e40e70dbe66f1359b347f08

File: Networking/NetworkingTests/Remote/GenerativeContentRemoteTests.swift
Comment (selanthiraiyan) [reply Networking/NetworkingTests/Remote/GenerativeContentRemoteTests.swift]: This checks that JWT is not requested again until "invalid token error" is received. Added comment in https://github.com/woocommerce/woocommerce-ios/commit/1a1561278eae0f2e4e40e70dbe66f1359b347f08

Same as https://github.com/woocommerce/woocommerce-ios/pull/10218#discussion_r1263549659

**Comment by selanthiraiyan** [comment]
@itsmeichigo Paolo confirmed that it is better to have separate feature names for each platform here pe5sF9-1Il-p2#comment-2536

Kindly add your comments to the P2 if you have any. Thank you!

**Pull request merged**
